org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/GitCredentialsService.java

```diff
@@ -0,0 +1,355 @@
+	 * in the secure storage, due limitations of using actual repository url as
+	 * node value in secure storage. Secure storage limits the length of node
+	 * value and also limits the use of special characters.
+	 *
+	 * @param url
+	 * @return MD5 hash value of the URL
+	 */
+	public static String getMD5HashForUrl(String url) {
+		String hashedURL = null;
+
+		if (url != null) {
+			try {
+				MessageDigest messageDigest = MessageDigest.getInstance("MD5"); //$NON-NLS-1$
+				messageDigest.update(url.getBytes());
+				byte[] digest = messageDigest.digest();
+				hashedURL = DatatypeConverter.printHexBinary(digest).toUpperCase();
+			} catch (NoSuchAlgorithmException e) {
+				AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+			}
+		}
+		return hashedURL;
+
+	}
+
+	/**
+	 * @param dialog
+	 *            Display longText in the DialogTray of the given dialog
+	 */
+	public static void handleExceptionAndDisplayInDialogTray(Exception e, TrayDialog dialog) {
+		if (e instanceof ResourceException && GitCredentialsService.isAuthenticationIssue((ResourceException) e)) {
+
+			final String longText = ((ResourceException) e).getExceptionLongtext();
+			if (longText != null) {
+				Display.getDefault().asyncExec(new Runnable() {
+
+					@Override
+					public void run() {
+						showErrorInDialogTray(longText, dialog);
+					}
+
+					public void showErrorInDialogTray(String longtext, TrayDialog dialog) {
+						GC gc = new GC(dialog.getShell());
+						gc.setFont(JFaceResources.getDialogFont());
+						FontMetrics fontMetrics = gc.getFontMetrics();
+						gc.dispose();
+
+						if (dialog.getTray() != null) {
+							dialog.closeTray();
+						}
+
+						DialogTray tray = new DialogTray() {
+							@Override
+							protected Control createContents(Composite parent) {
+								Composite root = new Composite(parent, SWT.NO_FOCUS);
+								GridLayoutFactory.fillDefaults().spacing(0, 0).applyTo(root);
+								Browser browser = BrowserUtil.createBrowserBasedOnPreferenceSettings(root);
+								browser.setText(addStyleSheetTo(longtext));
+								int xHint = Dialog.convertHorizontalDLUsToPixels(fontMetrics, 200);
+								GridDataFactory.fillDefaults().grab(true, true).hint(xHint, SWT.DEFAULT).applyTo(browser);
+								return root;
+							}
+
+							private String addStyleSheetTo(String htmlPage) {
+								IHtmlUtil htmlUtil = AbapCoreUi.getInstance().getHtmlUtil();
+								return htmlUtil.getHtmlPage(htmlPage, htmlUtil.getDefaultStyleSheetForDialogs());
+							}
+						};
+
+						dialog.openTray(tray);
+					}
+
+				});
+			}
+
+		}
+
+	}
+
+
+	public static void handleExceptionAndDisplayInErrorDialog(Exception e, Shell shell) {
+		//invalid credentials : this condition check is only valid from 2002
+		if (e instanceof ResourceException && GitCredentialsService.isAuthenticationIssue((ResourceException) e)) {
```

Comment:
Can we have this method as a generic error handler and not specifically for authorization related issues?
I mean we do not need the condition check right?

Reply:
Removed the check for AuthorizationIssue.